### PR TITLE
FEAT : Add SQS Publisher

### PR DESCRIPTION
**Pull request by saveroz**
**Changes**:
- add inquiry 
- add payment
- add sqs publisher

### CODE REVIEW

File: internal/postgres/export_test.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package postgres
+
+var PaymentCodeRepoTest Repository
```
Comment (saveroz) [inline internal/postgres/export_test.go:3]: i'm not sure if this is the correct way to export the variable used in [initialize repo ](https://github.com/saveroz/saveroz-golang-training-beginner/blob/beginner-part-5-SQS/internal/postgres/postgres_suite_test.go#L25-L39)for integration test.

File: .env.example [hunk 2]
```diff
@@ -2,4 +2,9 @@ POSTGRES_DB=postgres://user:password@postgres-test:5432/paymentcodedb?sslmode=di
 POSTGRES_DB_TEST=postgres://user:password@localhost:54320/paymentcodedb?sslmode=disable
 LOCAL_POSTGRES_DB=postgresql://postgres:postgres@localhost:5432/test?sslmode=disable
 REDIS_HOST=redis:6379
-REDIS_HOST_TEST=localhost:6379
\ No newline at end of file
+REDIS_HOST_TEST=localhost:6379
+AWS_REGION=ap-southeast-2
+AWS_ENDPOINT = localstack:4566
+AWS_ENDPOINT_TEST= http://localhost:4566
+AWS_SQS_URL = localstack:4566/000000000000/payment-queue
```
Comment (haveiss) [inline .env.example:9]: a bit typo, no need space before/after `=`
```suggestion
AWS_ENDPOINT=localstack:4566
AWS_ENDPOINT_TEST=http://localhost:4566
AWS_SQS_URL=localstack:4566/000000000000/payment-queue
```

File: app/main.go [hunk 3]
```diff
@@ -27,25 +37,39 @@ func main() {
...
 	if err != nil {
 		log.Fatal(err)
 	}
-	repo := postgres.New(db)
-	service := paymentcodeservice.New(repo, redisRepo)
+	paymentCodeRepo := postgres.New(db)
+	inquiryRepo := postgres.NewInquiryRepository(db)
+	paymentRepo := postgres.NewPaymentRepository(db)
+
+    paymentPublisher := paymentSqs.NewPaymentPublisher(
```
Comment (haveiss) [inline app/main.go:58]: [nit] i think your formatter not configured properly

can run this manually too before commit to format entire tree

```
go fmt ./...
```

File: internal/postgres/payment.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+		if isDuplicateError {
+			errCode = "DUPLICATE_PAYMENT_ERROR"
+		}
+
+		standardError = &standarderror.StandardError{
+			ErrorCode:    errCode,
+			ErrorMessage: errMessage,
+			StatusCode:   500,
+		}
```
Comment (haveiss) [inline internal/postgres/payment.go:59]: repository should be agnostic about delivery layer, for this example no need to specify statuscode in here,
could abstract something like this
https://github.com/xendit/cash-payment-service/blob/master/pkg/errors/duplicate_error.go

File: internal/postgres/payment.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	insertedRow, err := result.RowsAffected()
+
+	if insertedRow != 1 || err != nil {
+		standardError = &standarderror.StandardError{
+			ErrorCode:    errCode,
+			ErrorMessage: errMessage,
+			StatusCode:   500,
+		}
```
Comment (haveiss) [inline internal/postgres/payment.go:70]: see my comment above

File: internal/postgres/paymentcode.go [hunk 6]
```diff
@@ -187,3 +187,41 @@ func (r Repository) Expire() (standardError *standarderror.StandardError) {
...
+		&paymentCode.CreatedAt,
+		&paymentCode.UpdatedAt,
+	)
+
+	if err != nil {
+		standardError = &standarderror.StandardError{
+			ErrorCode:    "PAYMENT_CODE_NOT_FOUND_ERROR",
+			ErrorMessage: "payment code is not found",
+			StatusCode:   404,
+		}
```
Comment (haveiss) [inline internal/postgres/paymentcode.go:222]: see my previous comment

File: internal/postgres/postgres_suite_test.go [hunk 7]
```diff
@@ -2,12 +2,38 @@ package postgres_test
...
+
+	"github.com/saveroz/saveroz-golang-training-beginner/internal/postgres"
 )
 
+
 func TestPostgres(t *testing.T) {
 	gomega.RegisterFailHandler(ginkgo.Fail)
 	ginkgo.RunSpecs(t, "Postgres Suite")
 }
+const TIMEOUT =5000
```
Comment (haveiss) [inline internal/postgres/postgres_suite_test.go:23]: [nit] same as previous comment, i think your formatter is not running well

File: internal/sqs/paymentpublisher.go [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+	_, err := p.sqs.SendMessage(&sqs.SendMessageInput{
+		MessageBody: aws.String(string(messageBody)),
+		QueueUrl:    &p.queueUrl,
+	})
+	if err != nil {
+		standardError := &standarderror.StandardError{
+			ErrorCode:    "PUBLISH_PAYMENT_ERROR",
+			ErrorMessage: "fail to publish message",
+			StatusCode:   500,
+		}
```
Comment (haveiss) [inline internal/sqs/paymentpublisher.go:35]: see my previous comment about error abstraction

File: app/main.go (hunk 3 above)
Comment (saveroz) [reply app/main.go:58]: hmm weird in the vscode, the codes is placed properly. 
anyway thanks for the advice bro haveiss :+1:

File: .env.example (hunk 2 above)
Comment (saveroz) [reply .env.example:9]: thanks :pray:

File: internal/postgres/postgres_suite_test.go (hunk 7 above)
Comment (saveroz) [reply internal/postgres/postgres_suite_test.go]: formatted the code :heavy_check_mark:

File: internal/postgres/payment.go (hunk 4 above)
Comment (saveroz) [reply internal/postgres/payment.go:59]: thanks for the comment. 
I have a question, do I just need to remove the status code here and mapping the status code in the controller ? or do I need to strictly follow the example you gave?

Comment (haveiss) [reply internal/postgres/payment.go:59]: No need to strictly follow the example, just need to abstract it away from repository for the status code
this is another example
https://github.com/bxcodec/go-clean-arch/blob/9e174b8b0bbdfbab69bc293bb2905b2bb622155c/article/delivery/http/article_handler.go#L118-L134

Comment (saveroz) [reply internal/postgres/payment.go]: thanks pak i've update it

**Review by haveiss — APPROVED** [review]

**Pull request merged**